#!/usr/bin/env python
PACKAGE = "flipper_control"



from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#		      Name				        			Type		Reconfiguration level   	   		   	Description	  	Default	   	 Min		 Max
# ------- B O T H . F L I P P E R S ------------------------------------------------------------------------------------------------------------------#

gen.add("R", double_t, 0, "A double parameter",    0.082, 0.0,  0.1)
gen.add("r", double_t, 0, "A double parameter",    0.0375, 0.0,  0.05)
gen.add("L", double_t, 0, "A double parameter",    0.23, 0.0,  0.4)
gen.add("resolution", double_t, 0, "A double parameter",    0.06, 0.0,  0.4)
gen.add("flipperWidth", double_t, 0, "A double parameter",    0.1, 0.0,  0.3)
gen.add("trackLength", double_t, 0, "A double parameter",    0.6, 0.0,  1.0)
gen.add("TracksBaseLinkDist", double_t, 0, "A double parameter",    0.275, 0.0,  0.5)
gen.add("cropeMapLength", double_t, 0, "A double parameter",    2, 0,  4.0)
gen.add("fitPlaneLength", double_t, 0, "A double parameter",    0.8, 0,  4.0)

#


exit(gen.generate(PACKAGE, "flipper_control", "FlipperControl"))
